server/src/main/java/org/elasticsearch/index/query/BoolPrefixQueryBuilder.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.search.Queries;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.search.MatchQuery;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * The boolean prefix query analyzes the input text and creates a boolean query containing a Term query for each term, except
+ * for the last term, which is used to create a prefix query
+ */
+public class BoolPrefixQueryBuilder extends AbstractQueryBuilder<BoolPrefixQueryBuilder> {
```

Comment:
Should we rename to `MatchBoolPrefixQueryBuilder` for consistency ?

Reply:
I think the original user-facing name you mentioned was `match_prefix` which would be consistent with the other match queries. And the behavior would be what you'd expect based on an understanding of what the other match queries do

Reply:
Agreed but I think I prefer `match_bool_prefix` which is more verbose but also more consistent with the other names.

Reply:
Sounds good let's go with `match_bool_prefix`